### PR TITLE
Tensorflow: export LD_LIBRARY_PATH so protoc.bin command works when generating C and C++ files from proto files.

### PR DESCRIPTION
**Pull request by gartung**
Noticed the build of tensorflow:tensorflow would fail when invoking protoc because a newer libstdc++ was not found in the environment. Exporting LD_LIBRARY_PATH allowed the shell invoked by bazel to have the correct LD_LIBRARY_PATH set.

### CODE REVIEW

**Comment by gartung** [comment]
test with cms-exernals/tensorflow#7

File: tensorflow-sources.file [hunk 1]
```diff
@@ -73,6 +73,7 @@ export TF_NEED_ROCM=0
 export TF_NEED_TENSORRT=0
 export TEST_TMPDIR=%{_builddir}/build
 export TF_CMS_EXTERNALS="%{_builddir}/cms_externals.txt"
+export LD_LIBRARY_PATH
```
Comment (smuzaffar) [inline tensorflow-sources.file:76]: `LD_LIBRARY_PATH` should already be exported via the init.sh files? why you think it is not exported?

Comment (gartung) [reply tensorflow-sources.file:76]: I am trying to reproduce this build for Fermilab, but I keep getting errors like the one below. This export is probably not needed but there are a few places where I am trying to add use_default_shell_env in the tensorflow build rules.
```
ERROR: /scratch/gartung/products/tensorflow/v2_3_1/build/Linux64bit+3.10-2.17-e19-prof/tensorflow-2.3.1/tensorflow/core/data/service/BUILD:311:1: Action tensorflow/core/data/service/dispatcher.grpc.pb.h failed (Exit 1): protoc.bin failed: error executing command
  (cd /scratch/gartung/products/tensorflow/v2_3_1/build/Linux64bit+3.10-2.17-e19-prof/build/0dcd3170592590a4a5a279ac52482acc/execroot/org_tensorflow && \
  exec env - \
  bazel-out/k8-opt/bin/external/com_google_protobuf/protoc.bin '--plugin=protoc-gen-PLUGIN=bazel-out/k8-opt/bin/external/com_github_grpc_grpc/src/compiler/grpc_cpp_plugin' '--PLUGIN_out=generate_mock_code=true:bazel-out/k8-opt/bin' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=.' '--proto_path=bazel-out/k8-opt/bin' tensorflow/core/data/service/dispatcher.proto)
Execution platform: @local_execution_config_platform//:platform
bazel-out/k8-opt/bin/external/com_google_protobuf/protoc.bin: /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by bazel-out/k8-opt/bin/external/com_google_protobuf/protoc.bin)
bazel-out/k8-opt/bin/external/com_google_protobuf/protoc.bin: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /scratch/gartung/products/protobuf/v3_12_3/Linux64bit+3.10-2.17-e19/lib/libprotobuf.so.23)
bazel-out/k8-opt/bin/external/com_google_protobuf/protoc.bin: /lib64/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /scratch/gartung/products/protobuf/v3_12_3/Linux64bit+3.10-2.17-e19/lib/libprotobuf.so.23)
bazel-out/k8-opt/bin/external/com_google_protobuf/protoc.bin: /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /scratch/gartung/products/protobuf/v3_12_3/Linux64bit+3.10-2.17-e19/lib/libprotobuf.so.23)
bazel-out/k8-opt/bin/external/com_google_protobuf/protoc.bin: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /scratch/gartung/products/protobuf/v3_12_3/Linux64bit+3.10-2.17-e19/lib/libprotobuf.so.23)
bazel-out/k8-opt/bin/external/com_google_protobuf/protoc.bin: /lib64/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /scratch/gartung/products/protobuf/v3_12_3/Linux64bit+3.10-2.17-e19/lib/libprotoc.so.23)
bazel-out/k8-opt/bin/external/com_google_protobuf/protoc.bin: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /scratch/gartung/products/protobuf/v3_12_3/Linux64bit+3.10-2.17-e19/lib/libprotoc.so.23)
bazel-out/k8-opt/bin/external/com_google_protobuf/protoc.bin: /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /scratch/gartung/products/protobuf/v3_12_3/Linux64bit+3.10-2.17-e19/lib/libprotoc.so.23)
bazel-out/k8-opt/bin/external/com_google_protobuf/protoc.bin: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /scratch/gartung/products/protobuf/v3_12_3/Linux64bit+3.10-2.17-e19/lib/libprotoc.so.23)
Target //tensorflow/tools/pip_package:build_pip_package failed to build
ERROR: /scratch/gartung/products/tensorflow/v2_3_1/build/Linux64bit+3.10-2.17-e19-prof/tensorflow-2.3.1/tensorflow/core/data/service/BUILD:235:1 Action tensorflow/core/data/service/dispatcher.grpc.pb.h failed (Exit 1): protoc.bin failed: error executing command

```

Comment (gartung) [reply tensorflow-sources.file:76]: See https://github.com/cms-externals/tensorflow/pull/7

Comment (gartung) [reply tensorflow-sources.file:76]: LD_LIBRARY_PATH was not being set in the action environment by the configure script so I add this line by hand after running configure
```
echo "build --action_env LD_LIBRARY_PATH=\"${LD_LIBRARY_PATH}\"" >>.tf_configure.bazelrc
```

Comment (smuzaffar) [reply tensorflow-sources.file:76]: We have patched bazel  to use `True` as defaultvalue for use_default_shell_env parameter.

**Pull request closed without merging**